account_financial_report/wizard/general_ledger_wizard.py

```diff
@@ -108,7 +108,7 @@ def on_change_account_range(self):
             )
             if self.company_id:
                 self.account_ids = self.account_ids.filtered(
-                    lambda a: a.company_id == self.company_id
+                    lambda a: self.company_id in a.company_ids
```

Comment:
I think we could include company_ids in the search (with a if to increment the domain) and this way get rid of the filtered.
But it is more a refactore than a fix, and I believe we should do a large refactore of this module to be able to get a report for multiple companies (but necessarly all). It could be done at this time.